src/utils/index.js

```diff
@@ -24,20 +25,38 @@ var writeErrorHead = function( res, statusCode ) {
     'X-Content-Type-Options': config.defaultHeaders[ 'X-Content-Type-Options' ],
     'Content-Security-Policy': config.defaultHeaders[ 'Content-Security-Policy' ],
     'Strict-Transport-Security': config.defaultHeaders[ 'Strict-Transport-Security' ]
-  } );
+  };
+
+  if(contentType) {
+    headers['Content-Type'] = contentType;
+  }
+
+  res.writeHead( statusCode, headers );
 };
 
 /*
- * respond with a 404 response
+ * for backwards compatibility with legacy clients, respond with a placeholder image instead of a normal 404 response
  *
  * @param   object    res     http response object
  * @param   string    msg     404 message string
  * @param   object    url     url object
  */
-var fourOhFour = function( res, msg, url ) {
+var fourOhFour = function( res, msg, url, callback ) {
```

Comment:
By callback, I didn't mean the function takes a callback. There's no reason to. 

When you execute readFile, the anonymous function you attach as the second parameter IS the callback.